lib/jwt/jwk_test.go

```diff
@@ -34,3 +38,34 @@ func TestMarshalJWK(t *testing.T) {
 	// Required for integrating with AWS OpenID Connect Identity Provider.
 	require.Equal(t, "sig", jwk.Use)
 }
+
+func TestKeyIDHasConsistentOutputForAnInput(t *testing.T) {
```

Comment:
It'd be nice to see a test which takes a known public key and asserts that it produces a fixed KeyID - that'd let us ensure that we don't accidentally introduce a regression that causes the key ID in an already issued JWT to mismatch the key ID in the jwks.

Reply:
Good idea, added in c44c7c2d1441883c6ee8247669e39e136baad3a7.